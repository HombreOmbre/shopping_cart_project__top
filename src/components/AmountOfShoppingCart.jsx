import { useContext, useMemo } from "react";
import { ShoppingCartContext } from "../context/ShoppingCartContext.jsx";
import styles from "../styles/AmountOfShoppingCart.module.scss";

export const AmountOfShoppingCart = () => {
    const { shoppingCart } = useContext(ShoppingCartContext);
    const amount = useMemo(() => {
        return shoppingCart.reduce((allPrice, product) => {
            return allPrice + product.getProductPrice();
        }, 0);
    },[shoppingCart]);

    return (
        <div className={styles.container}>
            <div className={styles.amountTitle}>
                Amount:
            </div>
            <div className={styles.amountValue}>
                ${amount.toFixed(2)}
            </div>
        </div>
    );
}