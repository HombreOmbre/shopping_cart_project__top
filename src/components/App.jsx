import { useProducts } from "../customHooks/useProducts.jsx";

export const App = () => {
  const { products, error, loading } = useProducts();

  if (error) return <div>{error}</div>;
  if (loading) return <div>Loading...</div>;

  return (
      <ul>
      {
        products.length > 0 ?
             products.map((product) => {
               return (
                   <li key={product.id}>{product.title}</li>
               );
             }) : null
      }
      </ul>
  );
}
