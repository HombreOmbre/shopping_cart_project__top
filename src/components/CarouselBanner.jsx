import Carousel from "react-bootstrap/Carousel";
import 'bootstrap/dist/css/bootstrap.css';
import { useState } from "react";
import { Link } from "react-router-dom";
import { bannersData } from "../assets/banners.js";
import styles from "../styles/CarouselBanner.module.scss";

export const CarouselBanner = () => {
    const [index, setIndex] = useState(0);

    const handleSelect = (selectedIndex) => {
        setIndex(selectedIndex);
    }

    return (
        <div className={styles.mainContainer}>
            <Carousel activeIndex={index} onSelect={handleSelect} fade controls={false} indicators={false}  interval={1500} className={styles.container}>
                {
                    bannersData.map(item => (
                        <Carousel.Item key={item.id}>
                            <Link to="/">
                                <img src={item.img} alt={item.alt} className={styles.img} />
                            </Link>
                        </Carousel.Item>
                    ))
                }
            </Carousel>
        </div>
    );
}
