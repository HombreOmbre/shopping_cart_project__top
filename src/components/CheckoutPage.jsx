import { useContext } from "react";
import { ShoppingCartContext } from "../context/ShoppingCartContext.jsx";
import styles from "../styles/CheckoutPage.module.scss";
import { ShoppingCartItem } from "./ShoppingCartItem.jsx";

export const CheckoutPage = () => {
    const { shoppingCart } = useContext(ShoppingCartContext);
    return (
        <div className={styles.container}>

            {
                shoppingCart.length > 0 ? (
                    <>
                        <h2 className={styles.titleHeader}>Shopping cart</h2>
                        {
                            shoppingCart.map((product) => (
                                <ShoppingCartItem
                                    key={product.getId()}
                                    product={product}
                                />
                            ))
                        }

                    </>
                ) :
                <h2 className={styles.infoTxt}>There is no products in the shopping cart</h2>
            }
        </div>
    );
}