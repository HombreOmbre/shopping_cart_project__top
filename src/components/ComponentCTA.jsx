import { useRef, useState } from "react";
import styles from "../styles/ComponentCTA.module.scss";

export const ComponentCTA = () => {
    const emailInp = useRef(null);
    const [send, setSend] = useState(false);
    const [isValid, setIsValid] = useState(false);

    const submitHandler = (e) => {
        e.preventDefault();

        const regex = new RegExp(/[a-zA-Z\d]+@[a-zA-Z\d]+\.[a-zA-Z\d]{2,}/);

        if (emailInp.current.value.length === 0) {
            setIsValid(true);
            return;
        }

        if (!regex.test(emailInp.current.value.trim())) {
            setIsValid(true);
            return;
        }

        setSend(true);
        setIsValid(false);
    }

    return (
        <div className={styles.container}>
            <h2 className={styles.title}>Stay in touch</h2>
            <div className={styles.txtCTA}>
                Lorem ipsum dolor, sit amet consectetur adipisicing elit. Earum id, iure consectetur et error hic!
            </div>
            {!send ?
                <form className={styles.form}>
                    <input type="email" defaultValue='' ref={emailInp} className={styles.email} placeholder="Enter your email"/>
                    {
                        isValid &&
                        <span className={styles.errorEmail}>Email is invalid. Please try example@example.com</span>
                    }
                    <button type="submit" className={styles.submit} onClick={submitHandler}>Subscribe</button>
                </form> :
                <div className={styles.subscribe}>Thank You for subscribe!</div>
            }
        </div>
    );
}