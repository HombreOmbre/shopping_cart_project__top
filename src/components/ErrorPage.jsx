import styles from "../styles/ErrorPage.module.scss";

export const ErrorPage = () => {
    return (
        <div className={styles.errorPage}>
            <h1 className={styles.header}>Ooops!</h1>
            <p className={styles.errorTxt}>Something gone wrong...</p>
        </div>
    )
}