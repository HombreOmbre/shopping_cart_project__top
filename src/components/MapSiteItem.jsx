import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import styles from "../styles/MapSiteItem.module.scss";

export const MapSiteItem = ({ title, linksToSite }) => {
    return (
        <div className={styles.mapSiteContainer}>
            <h2 className={styles.title}>{title}</h2>
            <ul className={styles.links}>
                {linksToSite.map((siteLink, index) => (
                    <li className={styles.link} key={index}>
                        <Link to="/" className={styles.linkComp}>{siteLink}</Link>
                    </li>
                ))}
            </ul>
        </div>
    );
}

MapSiteItem.propTypes = {
    title: PropTypes.string.isRequired,
    linksToSite: PropTypes.arrayOf(PropTypes.string).isRequired,
}