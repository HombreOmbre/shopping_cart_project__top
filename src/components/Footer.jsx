import { ComponentCTA } from "./ComponentCTA.jsx";
import { MapSiteItem } from "./MapSiteItem.jsx";
import { mapSites } from "../assets/mapSites.js";
import styles from "../styles/Footer.module.scss";

export const Footer = () => {
    return (
        <footer className={styles.footer}>
            <div className={styles.footerContainer}>
                <div className={styles.footerContent}>
                    <div className={styles.title}>
                        SHOPPEROOO
                    </div>
                    <div className={styles.desc}>
                        Lorem ipsum dolor, sit amet consectetur adipisicing elit. Incidunt consequuntur amet culpa cum itaque neque.
                    </div>
                </div>
                <div className={styles.footerMapSite}>
                    {
                        mapSites.map((map, index) => (
                            <MapSiteItem key={index} title={map.title} linksToSite={map.links} />
                        ))
                    }
                    <ComponentCTA />
                </div>
                <div className={styles.footerRights}>
                    Copyright Â© HombreOmbre {new Date().getFullYear()}. All rights reserved.
                </div>
            </div>
        </footer>
    );
}