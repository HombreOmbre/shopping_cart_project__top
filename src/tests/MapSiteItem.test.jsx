import { render, screen } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import { MapSiteItem } from '../components/MapSiteItem';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom';

describe('MapSiteItem Component', () => {
    it('renders correctly with given title and links', () => {
        const props = {
            title: 'Useful Links',
            linksToSite: ['Home', 'About Us', 'Contact', 'Services'],
        };

        render(
            <MemoryRouter>
                <MapSiteItem title={props.title} linksToSite={props.linksToSite} />
            </MemoryRouter>
        );

        expect(screen.getByText('Useful Links')).toBeInTheDocument();

        props.linksToSite.forEach((linkText) => {
            expect(screen.getByText(linkText)).toBeInTheDocument();
        });
    });

    it('matches the snapshot', () => {
        const props = {
            title: 'Quick Links',
            linksToSite: ['Home', 'FAQ', 'Privacy Policy', 'Terms of Service'],
        };

        const { asFragment } = render(
            <MemoryRouter>
                <MapSiteItem title={props.title} linksToSite={props.linksToSite} />
            </MemoryRouter>
        );

        expect(asFragment()).toMatchSnapshot();
    });
});