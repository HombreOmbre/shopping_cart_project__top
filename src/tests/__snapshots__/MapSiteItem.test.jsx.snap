// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MapSiteItem Component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="_mapSiteContainer_7314e2"
  >
    <h2
      class="_title_7314e2"
    >
      Quick Links
    </h2>
    <ul
      class="_links_7314e2"
    >
      <li
        class="_link_7314e2"
      >
        <a
          class="_linkComp_7314e2"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="_link_7314e2"
      >
        <a
          class="_linkComp_7314e2"
          href="/"
        >
          FAQ
        </a>
      </li>
      <li
        class="_link_7314e2"
      >
        <a
          class="_linkComp_7314e2"
          href="/"
        >
          Privacy Policy
        </a>
      </li>
      <li
        class="_link_7314e2"
      >
        <a
          class="_linkComp_7314e2"
          href="/"
        >
          Terms of Service
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
